### PR TITLE
CVSL-1910-filtering-cases-with-out-delius-records

### PR DESCRIPTION
**Pull request by nishanthmahasamudram**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadService.kt [hunk 1]
```diff
@@ -151,7 +151,7 @@ class CaCaseloadService(
         licenceVersionOf = licence?.versionOf,
         name = "${licence?.forename} ${licence?.surname}",
         prisonerNumber = licence?.nomisId!!,
-        releaseDate = releaseDate!!,
+        releaseDate = releaseDate,
```
Comment (nishanthmahasamudram) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadService.kt:154]: removed not necessary null check.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadService.kt [hunk 2]
```diff
@@ -454,7 +454,7 @@ class CaCaseloadService(
             ),
           )
         } else {
-          ManagedCase(nomisRecord = prisoner.toPrisoner(), cvlFields = prisoner.toCaseloadItem().cvl)
+          null
```
Comment (nishanthmahasamudram) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadService.kt:457]: filtering out cases with no delius records

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadServiceTest.kt [hunk 3]
```diff
@@ -815,6 +821,46 @@ class CaCaseloadServiceTest {
           ),
         )
       }
+
+      @Test
+      fun `should filter out cases with no deliusRecord`() {
```
Comment (nishanthmahasamudram) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadServiceTest.kt:826]: Test to prove that.

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**